mkt/site/tests/__init__.py

```diff
@@ -430,6 +430,12 @@ def shortDescription(self):
     def _pre_setup(self):
         super(TestCase, self)._pre_setup()
 
+        # XXX See if we can remove this when we switch to Django 1.8.
+        # Some crud from the migration system gets stuck here and we have to
+        # kick it loose manually.
+        from mkt.webapps.models import AddonUser
+        UserProfile.addonuser_set.related.model = AddonUser
```

Comment:
When loading models, the migrations system creates fake versions of models that have all the same database schema as the regular ones; this is sufficient for its purposes but doesn't match the `isinstance` check done for foreign-key assignment. Some of this code was rewritten for 1.8 so I have hopes that it'll work better there (though our dependencies are far from ready to support 1.8 it seems).